sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,206 @@
+                                 "of bins. Received {0}, expected at least 2."
+                                 .format(n_bins))
+            return
+
+        n_bins = check_array(n_bins, dtype=int, ensure_2d=False)
+
+        if n_bins.ndim > 1 or n_bins.shape[0] != n_features:
+            raise ValueError("n_bins must be a scalar or array "
+                             "of shape (n_features_,).")
+
+        violating_indices = np.arange(n_features)[n_bins < 2]
+        if len(violating_indices) > 0:
+            indices = ", ".join(str(i) for i in violating_indices)
+            raise ValueError("Discretizer received an invalid number "
+                             "of bins at indices {}. Number of bins "
+                             "must be at least 2."
+                             .format(indices))
+
+    @staticmethod
+    def _check_categorical_features(cat_features, n_features):
+        if cat_features is None:
+            return
+
+        cat_features = check_array(cat_features, ensure_2d=False, dtype=int)
+        cat_features = column_or_1d(cat_features)
+
+        if len(set(cat_features)) != cat_features.shape[0]:
+            raise ValueError("Duplicate categorical column indices found.")
+
+        if np.all(cat_features >= 0) and np.all(cat_features < n_features):
+            return
+
+        raise ValueError("Invalid categorical feature index.")
+
+    def transform(self, X, y=None):
+        """Discretizes the data, ignoring any categorical features.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data containing features to be discretized.
+
+        Returns
+        -------
+        T : float array-like, shape (n_samples, n_features)
+            Categorical features will not be transformed.
+        """
+        check_is_fitted(self, ["mins_", "maxes_", "n_features_"])
+        X = check_array(X, dtype=float, ensure_2d=False)
+
+        if X.ndim == 1:
+            raise ValueError("Reshape your data.")
+
+        if X.shape[1] != self.n_features_:
+            raise ValueError("Transformed array does not have correct number "
+                             "of features. Expecting {}, received {}."
+                             .format(self.n_features_, X.shape[1]))
+
+        if self.categorical_features:
+            cat_features = np.asarray(sorted(self.categorical_features))
+        else:
+            cat_features = np.array([], dtype='int64')
+
+        # Manipulate the input attributes so categorical features can be
+        # ignored in transformation process. Also avoid divide by zero errors.
+        same_min_max = np.arange(self.n_features_)[self.maxes_ == self.mins_]
+        if len(same_min_max) != 0:
+            import warnings
+            warnings.warn("Fitted X contained continuous features at indices "
+                          "{}. These features will be discretized to the "
```

Comment:
I can probably change "These features will be discretized to the same value" to say "These features will be discretized to zero."